modules/header/header.service.js

```diff
@@ -0,0 +1,67 @@
+const Header = require('./header.model');
+const {
+  HEADER_ALREADY_EXIST,
+  HEADER_NOT_FOUND,
+} = require('../../error-messages/header.messages');
+
+class HeadersService {
+  async getAllHeaders() {
+    return await Header.find()
+  }
+
+  async getHeaderById(id) {
+    const foundHeader = await Header.findById(id);
+    if (foundHeader) {
+      return foundHeader;
+    }
+    throw new Error(HEADER_NOT_FOUND);
+  }
+
+  async updateHeader({ id, header }) {
+    if (await this.checkHeaderExist(header, id)) {
+      throw new Error(HEADER_ALREADY_EXIST);
+    }
+    const headerToUpdate = await Header.findById(id);
```

Comment:
If you just checking if `headerToUpdate` exists, then use the `lean()` method thus you reduce the response data, also I think this check should be provided first, since `checkHeaderExist` looks for `header` assuming that it certainly exists

Reply:
fixed